src/content/blog/2025-04-29-introducing-open-payments-php.mdx

```diff
@@ -0,0 +1,144 @@
+
+✅ Built for PHP 8.3+, using strict typing, PSR-4 autoloading, and Composer dependency management.
+
+✅ Clean, service-oriented architecture:
+
+- `Services/` : Interact with the Open Payments endpoints.
+- `Models/` : Represent API resources like grants, payments, and quotes.
+- `Validators/` : Validate API request payloads before sending.
+
+✅ Fully tested with PHPUnit, ready for production use.
+
+Here's what it looks like to request a grant:
+
+```php
+use OpenPayments\AuthClient;
+use OpenPayments\Config\Config;
+
+$config = new Config(
+    $WALLET_ADDRESS, $PRIVATE_KEY, $KEY_ID
+);
+$opClient = new AuthClient($config);
+
+$wallet = $opClient->walletAddress()->get([
+    'url' => $config->getWalletAddressUrl()
+]);
+
+$grant = $opClient->grant()->request(
+    [
+        'url' => $wallet->authServer
+    ],
+    [
+        'access_token' => [
+            'access' => [
+                [
+                    'type' => 'incoming-payment',
+                    'actions' => ['read', 'complete', 'create', 'list' ]
+                ]
+              ]
+        ],
+        'client' => $config->getWalletAddressUrl()
+    ]
+);
+```
+
+With just a few lines of code, PHP developers can now create grants, request quotes, initiate outgoing payments, or complete incoming payments — all fully compliant with Open Payments specifications (including generating HTTP signature headers).
+
+Detailed fully functional code snippets covering all endpoints can be found on the [Open Payments Php Snippets](https://github.com/interledger/open-payments-php-snippets/) repo.
+
+## 🛠️ How It Works: Inside the Library
+
+At its core, **Open Payments PHP** is designed around simplicity and modularity.
+The library is structured into clear layers, making it easy for developers to jump in, extend, or contribute:
+
+## 📁 Project Structure
+
+| Folder        | Purpose                                                             |
+| ------------- | ------------------------------------------------------------------- |
+| `Contracts/`  | Interface definitions (routes) for Services                         |
+| `Services/`   | Interact with Open Payments endpoints like grants, payments, quotes |
+| `Models/`     | Data models representing Open Payments resources                    |
+| `Validators/` | Input validation logic before making API calls                      |
+| `Traits/`     | Helpers like ID parsing and URL extraction                          |
+| `Utils/`      | Utility functions used for the http signature                       |
+
+Each service corresponds directly to parts of the Open Payments API.
+For example:
+
+- GrantService handles grant requests and continuations.
```

Comment:
Small thing, but can you make every item in this list either end with a period, or without? Doesn't matter which as long as we're consistent.